### PR TITLE
INT: Support module reexports

### PR DESCRIPTION
**Pull request by Undin**
These changes allow collecting items which are reachable via module reexports
```rust
mod foo {
    mod bar {
        pub mod baz {
            pub struct FooBar;
        }
    }

    pub use self::bar::baz; // <- reexported module
}

fn main() {
    let x = FooBar;  // <- now auto import can find correct import path `foo::baz::FooBar`
}
```

Note: these changes don't intent to support module reexports with alias and reexports with wildcard

### CODE REVIEW

**Comment by Undin** [comment]
@matklad or @vlad20012 r?

**Comment by vlad20012** [comment]
Can we also add heuristic that if we suggest a possible import from std, we hide imports from core/alloc/collections etc. This will also fix the failed test

**Comment by Undin** [comment]
@vlad20012 I thought about it but I want to implement it in separate PR

File: src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt [hunk 1]
```diff
@@ -58,23 +59,66 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
...
+        val visited: MutableSet<RsUseSpeck> = HashSet()
+
+        fun ImportItem.collectImportItems(): List<ImportItem> {
+            val importItems = mutableListOf(this)
+            val superMods = superMods.orEmpty()
+            superMods
+                // only public items can be reexported
+                .filter { it.isPublic }
+                .withIndex()
+                .forEach { (index, ancestorMod) ->
```
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt]: Indeed!
Some previous version used `.withIndex().flatMap()` and after switching to `forEach` I haven't collapsed it into `forEachIndexed`
Thanks)

**Pull request merged**